### ISSUE TITLE
Can't build in window using visual studio 2012 express cmd shell 

### ISSUE THREAD

**Issue by meiry** [mruby/mruby#2709]
Hello 
i try to build mruby in windows after following all the instructions i could find .
im using 
1. bison-2.4.1-bin
2. ruby 1.9.3p551 (2014-11-13) [i386-mingw32]
3.visual studio 2012 express 
4 win7  32 bit 
5. mruby master 

i open the Command prompted of visual studio 2012 express cd to mruby root dir 
and type make:
im getting this error:

```
D:\dev\cpp\2d\mruby\git\mruby>make
ruby ./minirake
(in D:/dev/cpp/2d/mruby/git/mruby)
CC    src/array.c -> build/host/src/array.obj
array.c
c1 : fatal error C1083: Cannot open source file: 'D:\dev\cppd\mruby\git\mruby\src\array.c': No such file or directory
array.c
c1 : fatal error C1083: Cannot open source file: 'D:\dev\cppd\mruby\git\mruby\src\array.c': No such file or directory
rake aborted!
Command Failed: [cl.exe /c /nologo /W3 /we4013 /Zi /MD /O2 /D_CRT_SECURE_NO_WARNINGS /DMRB_STACK_EXTEND_DOUBLING /DMRB_DEBUG /I"D:\dev\cppd\mruby
\git\mruby\include" /Fo"D:\dev\cppd\mruby\git\mruby\build\host\src\array.obj" "D:\dev\cppd\mruby\git\mruby\src\array.c"]

make: *** [all] Error 1


when i do :ruby minirake
im getting :
```

D:\dev\cpp\2d\mruby\git\mruby>ruby minirake
(in D:/dev/cpp/2d/mruby/git/mruby)
CC    src/array.c -> build/host/src/array.obj
array.c
c1 : fatal error C1083: Cannot open source file: 'D:\dev\cppd\mruby\git\mruby\src\array.c': No such file or directory
array.c
c1 : fatal error C1083: Cannot open source file: 'D:\dev\cppd\mruby\git\mruby\src\array.c': No such file or directory
rake aborted!
Command Failed: [cl.exe /c /nologo /W3 /we4013 /Zi /MD /O2 /D_CRT_SECURE_NO_WARNINGS /DMRB_STACK_EXTEND_DOUBLING /DMRB_DEBUG /I"D:\dev\cppd\mruby
\git\mruby\include" /Fo"D:\dev\cppd\mruby\git\mruby\build\host\src\array.obj" "D:\dev\cppd\mruby\git\mruby\src\array.c"]

```


i noticed there is somekind of path appending that going wrong see the "ccpd"
```

D:\dev\cppd\mruby\git\mruby\include

```
what im missing here ?
```

**Comment by sdottaka** [comment]
fixed by a99fb39

**Comment by meiry** [comment]
Thanks takashi for the quick fix But there are still error different onces i suspect it is again related to 
path with spaces here is the stdout when i run make .
the bison installed in this path :
```
C:\Program Files (x86)\GnuWin32\bin


```
D:\dev\cpp\2d\mruby\git\mruby>make
ruby ./minirake
(in D:/dev/cpp/2d/mruby/git/mruby)
CC    src/array.c -> build/host/src/array.obj
array.c
CC    src/backtrace.c -> build/host/src/backtrace.obj
backtrace.c
CC    src/class.c -> build/host/src/class.obj
class.c
CC    src/codegen.c -> build/host/src/codegen.obj
codegen.c
CC    src/compar.c -> build/host/src/compar.obj
compar.c
CC    src/crc.c -> build/host/src/crc.obj
crc.c
CC    src/debug.c -> build/host/src/debug.obj
debug.c
CC    src/dump.c -> build/host/src/dump.obj
dump.c
CC    src/enum.c -> build/host/src/enum.obj
enum.c
CC    src/error.c -> build/host/src/error.obj
error.c
CC    src/etc.c -> build/host/src/etc.obj
etc.c
CC    src/gc.c -> build/host/src/gc.obj
gc.c
CC    src/hash.c -> build/host/src/hash.obj
hash.c
CC    src/init.c -> build/host/src/init.obj
init.c
CC    src/kernel.c -> build/host/src/kernel.obj
kernel.c
CC    src/load.c -> build/host/src/load.obj
load.c
CC    src/numeric.c -> build/host/src/numeric.obj
numeric.c
CC    src/object.c -> build/host/src/object.obj
object.c
CC    src/pool.c -> build/host/src/pool.obj
pool.c
CC    src/print.c -> build/host/src/print.obj
print.c
CC    src/proc.c -> build/host/src/proc.obj
proc.c
CC    src/range.c -> build/host/src/range.obj
range.c
CC    src/state.c -> build/host/src/state.obj
state.c
CC    src/string.c -> build/host/src/string.obj
string.c
CC    src/symbol.c -> build/host/src/symbol.obj
symbol.c
CC    src/variable.c -> build/host/src/variable.obj
variable.c
CC    src/version.c -> build/host/src/version.obj
version.c
CC    src/vm.c -> build/host/src/vm.obj
vm.c
YACC  src/parse.y -> build/host/src/y.tab.c
m4: cannot open `Files': No such file or directory
m4: cannot open `(x86)\GnuWin32/share/bison': No such file or directory
m4: cannot open `C:\Program': No such file or directory
m4: cannot open `Files': No such file or directory
m4: cannot open `(x86)\GnuWin32/share/bison/m4sugar/m4sugar.m4': No such file or directory

**Comment by sdottaka** [comment]
I think there is no tool to generate visual studio project files for mruby.
So I created Visual Studio Project for my mrbgem manually. (https://github.com/sdottaka/mruby-win32ole/blob/master/vsproj/mruby.vcxproj).
This l might be helpful to you.

**Comment by sdottaka** [comment]
FYI I'm using bison in Git for Windows (C:\Program Files (x86)\Git\bin\bision.exe)

**Comment by meiry** [comment]
Thanks takashi

**Comment by sdottaka** [comment]
Yes, It may not be expected. It is a project that simply run 'ruby ./minirake' .
But the project is buildable and debugable.

**Comment by meiry** [comment]
Yes i saw that there are no sources or lib's included , 
but its good reference on how to use mruby . 
say if i want to create Visual Studio project , as im not coming from the ruby world 
what source and what stages i need to do to mimic the minirake builder ?

**Comment by sdottaka** [comment]
I'm not sure that this is the answer. I created Visual Studio project files for building mruby without Ruby. (https://github.com/sdottaka/mruby-vsproj)
But I actually don't recommend it because it is difficult to add new external mrbgems.

**Comment by matz** [comment]
Closing — VS 2012 Express build question from 2015, no follow-up. Modern Windows builds with MSVC use build_config/visualcpp.rb.
